ocs_ci/ocs/scale_lib.py

```diff
@@ -998,3 +1000,229 @@ def get_max_pvc_count():
         count += 1
     pvc_count = count * constants.SCALE_MAX_PVCS_PER_NODE
     return pvc_count
+
+
+def check_all_pod_reached_running_state_in_kube_job(
+    kube_job_obj, namespace, no_of_pod, timeout=30
+):
+    """
+    Function to check either bulk created PODs reached Running state using kube_job
+
+    Args:
+        kube_job_obj (obj): Kube Job Object
+        namespace (str): Namespace of PVC's created
+        no_of_pod (int): POD count
+
+    Returns:
+        pod_running_list (list): List of all PODs reached running state.
+
+    Asserts:
+        If not all POD reached Running state.
+
+    """
+
+    # Check all the POD reached Running state
+    pod_running_list, pod_not_running_list = ([] for i in range(2))
+    while_iteration_count = 0
+    while True:
+        # Get kube_job obj and fetch either all PODs are in Running state
+        # If not Running, adding those PODs to pod_not_running_list
+        job_get_output = kube_job_obj.get(namespace=namespace)
+        for i in range(no_of_pod):
+            if job_get_output["items"][0]["kind"] == constants.POD:
+                pod_type = constants.POD
+            else:
+                pod_type = None
+            if pod_type:
+                status = job_get_output["items"][i]["status"]["phase"]
+                logging.info(
+                    f"POD {job_get_output['items'][i]['metadata']['name']} status {status}"
+                )
+                if status != "Running":
+                    pod_not_running_list.append(
+                        job_get_output["items"][i]["metadata"]["name"]
+                    )
+            else:
+                # For DC config there is no Running status so checking it based on
+                # availableReplicas, basically this will be 1 if pod is running and
+                # the value will be 0 in-case of pod not in running state
+                status = job_get_output["items"][i]["status"]["availableReplicas"]
+                logging.info(
+                    f"DC Config {job_get_output['items'][i]['metadata']['name']} "
+                    f"available running pods {status}"
+                )
+                if not status:
+                    pod_not_running_list.append(
+                        job_get_output["items"][i]["metadata"]["name"]
+                    )
+
+        # Check the length of pod_not_running_list to decide either all PODs reached
+        # Running state, If not then wait for 30secs and re-iterate while loop
+        if len(pod_not_running_list):
+            time.sleep(timeout)
+            while_iteration_count += 1
+            # Breaking while loop after 10 Iteration i.e. after 30*10 secs of wait_time
+            # And if PODs are still not in bound state then there will be assert.
+            if while_iteration_count >= 10:
```

Comment:
i think this should depend on the number of pods, for instance,  1500 pods will take more time then 120 pods, and less then 10K pods

Reply:
Pods will be created using kube_job so in TC if we are creating 3000 PVC's and if we attach 30 PVCs to single pod then we will end up creating just 100 pods.
Will test one more time and update here to with results of 3000, 6000 PVCs creation.